File: uas-ppb-c030324023-mobile/lib/models/account.dart
class Account {
  final int id;
  final String role;
  final String? nisn;
  final String username;
  final String email;

  Account({
    required this.id,
    required this.role,
    this.nisn,
    required this.username,
    required this.email,
  });

  factory Account.fromJson(Map<String, dynamic> json) => Account(
        id: json['id'],
        role: json['role'],
        nisn: json['nisn'],
        username: json['username'],
        email: json['email'],
      );
}

File: uas-ppb-c030324023-backend/app/Models/Account.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Attributes\Hidden;
use Illuminate\Foundation\Auth\User as Authenticatable;

#[Fillable(['role', 'nisn', 'username', 'email', 'password'])]
#[Hidden(['password'])]
class Account extends Authenticatable
{
    protected $table = 'accounts';

    protected function casts(): array
    {
        return [
            'password' => 'hashed',
        ];
    }
}
